src/drivers/dw/dma.c

```diff
@@ -68,20 +65,11 @@ struct dw_dma_chan_data {
 
 	/* pointer data */
 	struct dw_dma_ptr_data ptr_data;
-
-	/* client callback function */
-	void (*cb)(void *data, uint32_t type, struct dma_cb_data *next);
-	/* client callback data */
-	void *cb_data;
-	/* callback type */
-	int cb_type;
-
-	struct dma *dma;
 };
 
 /* private data for DW DMA engine */
 struct dma_pdata {
-	struct dw_dma_chan_data chan[DW_MAX_CHAN];
+	struct dma_chan_data chan[DW_MAX_CHAN];
```

Comment:
Now it is a common struct, so don't need to be a part of dma private data.
It can be a dma struct member.

Reply:
DW_MAX_CHAN is kinda specific, and you can't use constants in the platform data to decide the size. So how should he do this?

Reply:
Or we could merge it with the one in platform data and allocate it dynamically based on number of channels assigned to the particular DMA.

Reply:
@tlauda Yeah, dynamic allocation is also an option. Let's just hope it doesn't lead to allocating more than 1024 bytes in one allocation (I have encountered issues with that while working on the almost-ready EDMA driver on i.MX).

Reply:
@tlauda I've extracted channels to dma struct and initialize them during dma_probe.